Source/WebKit/ModelProcess/cocoa/ModelProcessModelPlayerProxy.mm

```diff
@@ -352,11 +350,31 @@ static CGFloat effectivePointsPerMeter(CALayer *caLayer)
     RECALayerClientComponentSetShouldSyncToRemotes(layerComponent.get(), true);
 
     auto clientComponent = RECALayerGetCALayerClientComponent(m_layer.get());
-    auto rootEntity = REComponentGetEntity(clientComponent);
-    REEntitySetName(rootEntity, "WebKit:ClientComponentRoot");
+    auto clientComponentEntity = REComponentGetEntity(clientComponent);
+    REEntitySetName(clientComponentEntity, "WebKit:ClientComponentRoot");
     REEntitySetName(m_model->rootEntity(), "WebKit:ModelRootEntity");
-    REEntitySetParent(m_model->rootEntity(), rootEntity);
-    RENetworkMarkEntityMetadataDirty(rootEntity);
+
+    // FIXME: Clipping workaround for rdar://125188888 (blocked by rdar://123516357 -> rdar://124718417).
+    // middleEntity is required to add a clipping primitive that is independent from model's rootEntity.
+    // Adding the primitive directly to clientComponentEntity has no visual effect.
+    constexpr float clippingBoxHalfSize = 500; // meters
+    REPtr<REEntityRef> middleEntity = adoptRE(REEntityCreate());
```

Comment:
I'm fine with "middle" since I think this should go away once we move to clipping plane implementation.
But if this is to stay, I'd think "containerEntity" may be a better name than "middleEntity"

Reply:
Agree. Changed to `containerEntity`.